addons/stock/models/stock_move_line.py

```diff
@@ -1012,11 +1068,20 @@ def _check_destinations(self):
                 'target': 'new'
             }
 
-    def _put_in_pack(self):
-        package = self.env['stock.package'].create({})
-        package_type = self.move_id.packaging_uom_id.package_type_id
-        if len(package_type) == 1:
-            package.package_type_id = package_type
+    def _put_in_pack(self, package_id=False, package_type_id=False, package_name=False):
+        if package_id:
+            package = self.env['stock.package'].browse(package_id)
+        elif package_type_id:
+            package = self.env['stock.package'].create({
+                'name': package_name,
+                'package_type_id': package_type_id,
+            })
+        else:
+            package_vals = {'name': package_name}
```

Comment:
Is it possible for a `package_name` to not be passed? Since `stock.package.name` is required.

Reply:
Yes, the name will be generated if necessary in the `create()`.